### PR TITLE
Added the 'pipe' mechanism

### PR DESCRIPTION
**Pull request by jvdsande**
Completed the Emitter and Receiver components to implement wanted functionality
Added a parent class system to have multiple Emitter/Receiver pairs

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,25 +1,157 @@
 import React, {Component} from 'react'
 
-export class Emitter extends Component {
-  constructor(props) {
-    super(props)
-  }
+/*  The EmitterReceiver class aims to provide off-site rendering capabilities
+ *  to React
+ *  An instance of EmitterReceiver provides to Component classes, Emitter and
```
Comment (aymericbouzy) [inline src/index.js]: to -> two

File: src/index.js [hunk 2]
```diff
@@ -1,25 +1,157 @@
...
+          pipe.triggerReceivers()
+        }
+      }
+
+      // During its lifecycle, the component will need to give its content to
+      // the Receivers
+      add() {
+        // Get the content from the Emitter
+        let child = this.props.children
+        // Check if the Emitter acutally has content
```
Comment (aymericbouzy) [inline src/index.js]: actually

File: src/index.js [hunk 3]
```diff
@@ -1,25 +1,157 @@
...
+        if (child !== null && child !== undefined) {
```
Comment (aymericbouzy) [inline src/index.js]: code duplication

File: src/index.js [hunk 4]
```diff
@@ -1,25 +1,157 @@
...
+          child = child.length ? child : [child]  // Convert into Array
```
Comment (aymericbouzy) [inline src/index.js]: code duplication

**Review by aymericbouzy — CHANGES_REQUESTED** [review]
Bravo ! Il y a juste quelques corrections à faire :)

**Comment by jvdsande** [comment]
Requested changes implemented + minor logic improvement in the EmitterReceiver class

**Review by aymericbouzy — APPROVED** [review]

**Pull request closed without merging**